infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/transport/sample/GRPCLogReplicationClientChannelAdapter.java

```diff
@@ -380,6 +383,55 @@ public void onCompleted() {
         replicationReqObserverMap.get(Pair.of(sessionMsg, requestId)).onNext(request);
     }
 
+    private void sinkSideSessionInitialize(String nodeId, RequestMsg request) {
```

Comment:
This RPC can be blocking in nature. Currently its not.
But technically, since transport layer is pluggable for LR, this implementation can be anything.

But we would need the logic to retry in sessionManager which ends up calling this RPC.